lib/parse-route-string.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+/**
+ * Take a WP route string (with PCRE named capture groups), such as
+ * @module parseRouteString
+ */
+var util = require( 'util' );
+var _ = require( 'lodash' );
+function logFull( obj ) {
+	console.log( util.inspect( obj, {
+		colors: true,
+		depth: null
+	}) );
+}
+
+// All valid routes in API v2 beta 11
+var routes = require( './endpoint-response.json' ).routes;
+
+function forIn( obj, callbackFn ) {
+	Object.keys( obj ).forEach(function( key, idx ) {
+		callbackFn( obj[ key ], idx );
+	});
+}
+
+/*
+
+"/wp/v2/posts": {},
+"/wp/v2/posts/(?P<id>[\\d]+)": {},
+"/wp/v2/posts/(?P<parent>[\\d]+)/revisions": {},
+"/wp/v2/posts/(?P<parent>[\\d]+)/revisions/(?P<id>[\\d]+)": {},
+"/wp/v2/pages": {},
+"/wp/v2/pages/(?P<id>[\\d]+)": {},
+"/wp/v2/pages/(?P<parent>[\\d]+)/revisions": {},
+"/wp/v2/pages/(?P<parent>[\\d]+)/revisions/(?P<id>[\\d]+)": {},
```

Comment:
This shows an example of what the route definitions look like—currently, I parse out the named capture groups from the regexes and use them to construct a representation of each "level" of the tree, where a "level" is one segment of a URL between `/`s.

Here's what the generated tree looks like for the posts routes defined here:

![image](https://cloud.githubusercontent.com/assets/442115/13018531/0b0a2970-d198-11e5-808d-c7f526de42eb.png)

"dynamic" and "level" are inferred while walking the tree; validate methods are generated based on whether a given input matches the regex defined in a named capture group. "names" represents the list of names given to the same capture group in different routes under the same root resource type; the `name` could (and hopefully will) be used to auto-generate chaining methods to populate that segment of the URL.